server/repository/src/db_diesel/storage_connection.rs

```diff
@@ -93,13 +101,28 @@ impl StorageConnection {
 
         let con = &self.connection;
         let transaction_manager = con.transaction_manager();
-        transaction_manager.begin_transaction(con).map_err(|err| {
-            error!("Failed to begin tx: {:?}", err);
-            TransactionError::Transaction {
-                msg: format!("Failed to begin tx: {}", err),
-                level: current_level + 1,
+        if current_level == 0 {
+            // sqlite can only have 1 writer at a time, so to avoid concurrency issues,
+            // the first level transaction for sqlite, needs to run 'BEGIN IMMEDIATE' to start the transaction in WRITE mode.
```

Comment:
does this mean even for reads, we loose concurrency ?

Reply:
Yes, but only during a transaction. There is still concurrency for reads outside of transactions.
Essentially this means there's only one 'transaction' at a time.